padl/transforms.py

```diff
@@ -1552,16 +1634,63 @@ def _pd_get_keys(transforms):
             deduped_keys.append(new_name)
         return deduped_keys
 
+    def _pd_get_non_target_stage_idx(self, stage_used: bool):
+        """When a :class:`Compose` gets an Exception and has a non-failing stage which is a
+        :class:`Parallel` or a :class:`Rollout`, the index they contribute to track the
+        element that got the Exception on the entire :class:`Compose` is 1.
+
+        Example:
+            t = ((prep_1 >> batch) + (prep_2 >> batch)) >> forward_1
+            Let's suppose we get an error on `forward_1`, so the item that is failing on `t` is 1.
+            `forward_1` is the index 0 of `t.pd_forward`, then the element that fails on `t` is
+            t._pd_preprocess._pd_get_non_target_stage_idx() + 0 = 1 + 0 = 1
+
+        :param stage_used: *True* if stage used on the :class:`Compose`, else *False*
+        """
+        return 1
+
+    def _pd_get_target_stage_idx(self, is_entire_transform=None):
+        """When a :class:`Compose` gets an Exception and the failing stage is a :class:`Parallel`
+        or a :class:`Rollout`, the index they contribute with to track the element that got the
+        Exception on the entire :class:`Compose` is 0.
+
+        Example:
+            t = prep >> batch >> (forward_1 + forward_2)
+            Let's suppose we get an error on `(forward_1 + forward_2)`, so the item that is
+            failing on `t` is 2. `(forward_1 + forward_2)` is the index 0 of `t.pd_forward`,
+            then the element that fails on `t` is
+            2 + t._pd_forward._pd_get_stage_idx() = 2 + 0 = 2
+
+        :param is_entire_transform: *True* if *self* constitutes entirely an overlying
+            :class:`Transform`, else *False*
```

Comment:
I think this explanation is a little confusing because we have no definition of what an `overlying Transform` is. I think it would be clearer if you describe that `*self* is not the stage of a Pipeline` or something similar.

Reply:
I understand, and this is something that happens on other parts of the docs, but these methods are applied to stages of a transform, and I don't see a way to explain what these methods do without referring to the overlying transform.

Reply:
Yes, I agree that the explanation is not very clear. I am trying to rephrase it. If I understood correctly, @sjrl suggested to describe this method without mentioning that `self` is an stage of a larger `Transform`. If I don't mention this relation, I don't see a way to explain accurately what's the meaning of this method or what we are doing inside.

Reply:
Yes, I agree that the explanation is not very clear. I am trying to rephrase it, but it is a bit difficult without mentioning that `self` is an stage of a larger `Compose` (this is what @sjrl suggested, if I understood correctly). I am upgrading to a more minimal (and I think clearer) explanation. Let me know how you like it.

Reply:
I think the current comment is incorrect.
```
:param is_entire_transform: *True* if *self* is a part of a larger :class:`Transform`,
            else *False*
```
I think you meant 
```
*True* if *self* is not part of a larger :class:`Transform`, else *False*
```